bindings/java/src/org/sleuthkit/datamodel/BlackboardArtifact.java

```diff
@@ -384,6 +385,41 @@ public void addAttributes(Collection<BlackboardAttribute> attributes) throws Tsk
 		attrsCache.addAll(attributes);
 	}
 
+	/**
+	 * Adds a collection of attributes to this artifact in a single operation
+	 * (faster than adding each attribute individually). This function also
+	 * executes in the inserts within the transaction context.
+	 *
+	 * @param attributes		      The collection of attributes.
+	 * @param caseDbTransaction the transaction in the scope of which the
+	 *                          operation is to be performed, managed by the
+	 *                          caller
+	 *
+	 * @throws TskCoreException If an error occurs and the attributes were not
+	 *                          added to the artifact.
+	 */
+	public void addAttributes(Collection<BlackboardAttribute> attributes, final SleuthkitCase.CaseDbTransaction caseDbTransaction) throws TskCoreException {
+
+		if (attributes.isEmpty()) {
+			return;
+		}
+		if (caseDbTransaction == null) {
```

Comment:
The transaction should be a required argument, as it is currently documented, and we should throw an IllegalArgumentException if it is not provided. If a caller wants to have an implicit transaction, they can just call addAttributes(Collection<BlackboardAttribute> attributes) directly.